src/status_im/data_store/core.cljs

```diff
@@ -19,9 +20,16 @@
 (defn change-account [address new-account? encryption-key handler]
   (data-source/change-account address new-account? encryption-key handler))
 
-(defn- perform-transactions [transactions realm]
-  (data-source/write realm #(doseq [transaction transactions]
-                              (transaction realm))))
+(defn- perform-transactions [raw-transactions realm]
```

Comment:
could you specify the purpose of this change ?

Reply:
I was tracking down a bug where by it would try to write to a closed realm, turned out it was a different issue, but the problem still stand that writes to realm are asynchronous (they go through a core async queue), so some actions do depend on it to be completed, so we should only `logout` once the realm db transaction has been successfully written.